source/UF/ConnectedTypes.lagda

```diff
@@ -210,3 +254,176 @@ the identity type at one level below. We will assume univalence only when necess
+                                                  → f is n connected-map
+                                                  → ((y : Y) → P y)
+                                                   ≃ ((x : X) → P (f x))
+ dependent-equiv-from-truncated-fam-connected-map {_} {_} {𝓦} {X} {Y} {f} {P} {n}
+  P-trunc f-conn = e
+  where
+   e : ((y : Y) → P y) ≃ ((x : X) → P (f x))
+   e = ((y : Y) → P y)                                         ≃⟨ I ⟩
+       ((y : Y) → (fiber f y → P y))                           ≃⟨ II ⟩
+       ((y : Y) → (x : X) → (p : f x ＝ y) → P y)              ≃⟨ Π-flip' ⟩
+       ((x : X) → (y : Y) → (p : f x ＝ y) → P y)              ≃⟨ III ⟩
+       ((x : X) → P (f x))                                     ■
+    where
+     I = Π-cong fe fe (λ - → maps-from-connected-type-to-truncated-type-const
+                       (f-conn -) (P-trunc -))
+     II = Π-cong fe fe (λ - → curry-uncurry' fe fe)
+     III = Π-cong fe fe (λ - → ≃-sym (Yoneda-equivalence fe' (f -) P))
+   observation : ⌜ e ⌝ ＝ dprecomp P f
+   observation = refl
+
+ Lemma7-5-7-i : {X : 𝓤 ̇} {Y : 𝓥 ̇} {f : X → Y} {P : Y → 𝓦 ̇} {n : ℕ₋₂} 
+              → ((y : Y) → (P y) is n truncated)
+              → f is n connected-map
+              → is-equiv (dprecomp P f)
+ Lemma7-5-7-i {_} {_} {_} {X} {Y} {f} {P} {n} P-trunc f-conn =
+  ⌜⌝-is-equiv (dependent-equiv-from-truncated-fam-connected-map P-trunc f-conn)
+
+ Lemma7-5-7-ii : {X : 𝓤 ̇} {Y : 𝓥 ̇} {f : X → Y} {P : Y → 𝓦 ̇} 
+               → is-equiv (dprecomp P f)
+               → has-section (dprecomp P f)
+ Lemma7-5-7-ii {_} {_} {_} {_} {_} {f} {P} =
+  equivs-have-sections (dprecomp P f)
+
+ Lemma7-5-7-iii : {X : 𝓤 ̇} {Y : 𝓥 ̇} {f : X → Y} {n : ℕ₋₂} 
+                → ({𝓦 : Universe} {P : Y → 𝓦 ̇}
+                  → ((y : Y) → (P y) is n truncated)
+                  → has-section (dprecomp P f))
+                → f is n connected-map
+ Lemma7-5-7-iii {𝓤} {𝓥} {X} {Y} {f} {n} sec-from-trunc y = (c y , C)
+  where
+   Q : Y → 𝓤 ⊔ 𝓥 ̇
+   Q y = ∥ fiber f y ∥[ n ]
+   c' : ((x : X) → ∥ fiber f (f x) ∥[ n ])
+      → ((y : Y) → ∥ fiber f y ∥[ n ])
+   c' = section-of (dprecomp Q f) (sec-from-trunc (λ - → ∥∥ₙ-is-truncated))
+   c : (y : Y) → ∥ fiber f y ∥[ n ]
+   c = c' (λ - → ∣ (- , refl) ∣[ n ])
+   H' : (dprecomp Q f) ∘ c' ∼ id
+   H' = section-equation (dprecomp Q f)
+                         (sec-from-trunc (λ - → ∥∥ₙ-is-truncated))
+   H : (x : X) → c (f x) ＝ ∣ (x , refl) ∣[ n ]
+   H = happly' ((dprecomp Q f ∘ c') (λ - → ∣ (- , refl) ∣[ n ]))
+               (λ - → ∣ (- , refl) ∣[ n ]) (H' (λ - → ∣ (- , refl) ∣[ n ]))
+   C : (w : ∥ fiber f y ∥[ n ]) → c y ＝ w
+   C = ∥∥ₙ-ind (λ - → truncation-levels-are-upper-closed ∥∥ₙ-is-truncated
+                (c y) -)
+               C'
+    where
+     C' : ((x , p) : fiber f y) → c y ＝ ∣ (x , p) ∣[ n ]
+     C' (x , refl) = H x
+
+\end{code}
+
+We show that the canonical n-truncation map is n-connected.
```

Comment:
`the canonical n-truncation map` . I would say `the unit of the n-truncation` or just the `n-truncation map` and rename the result appropriately.